### PR TITLE
Allow Design Reviews to be edited before being finalized

### LINKED ISSUE #2510: DRC Debugs - Required/Optional Shouldn't be underlined
Required and optional shouldn't be underlined on the design review detail page

### LINKED ISSUE #2594: DRC Debugs - Sunday should be on the left (will take some refactoring)
### Description

Right now, the design review calendar goes Monday to Sunday, we want to change it to go, Sunday to Saturday left to right. This is probably going to be pretty tricky since it's going to involve delving into the custom calendar code, specifically the date part. 

### Acceptance Criteria

- [ ] The Design Review Calendar is Sunday to Saturday instead of Monday to Sunday at the top

### Proposed Solution

I think the way it currently is is based on the fact that a getDayOfTheWeek helper method for Dates indexes days of the week starting at Monday with 0, so you'll probably need to make a helper method to change the calculations so Sunday is the lowest index instead

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Changed the weird inversion of prop drilling to be top down which allowed for the date, starttime and end time to be edited along with the required users and everything else. 

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Code looks good. I tested it out and the cancel button on the edit drc page doesn't work. And for some reason the question doc link doesn't get saved. If I finalize a drc and add a link for the question doc, then click back on the drc it won't let me save unless I add a link for the question doc again. I don't know if that part is intended

**Comment by Peyton-McKee** [comment]
lol the cancel button must notve worked before too then

**Comment by Peyton-McKee** [comment]
> Code looks good. I tested it out and the cancel button on the edit drc page doesn't work. And for some reason the question doc link doesn't get saved. If I finalize a drc and add a link for the question doc, then click back on the drc it won't let me save unless I add a link for the question doc again. I don't know if that part is intended

good catches on the edits after finalize i didnt catch those

**Review by Aaryan1203 — DISMISSED** [review]
lgtm 🔥

File: src/backend/src/routes/design-reviews.routes.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ designReviewsRouter.post(
   body('isInPerson').isBoolean(),
   nonEmptyString(body('zoomLink')).isURL().optional(),
   nonEmptyString(body('location')).optional(),
-  nonEmptyString(body('docTemplateLink')).isURL().optional(),
```
Comment (walker-sean) [inline src/backend/src/routes/design-reviews.routes.ts:41]: Why remove this

**Review by walker-sean — CHANGES_REQUESTED** [review]
Editing allows you to select an invalid start & end time, which leads to the scheduled time being NaN

Also this is what it looks like when you select the week of an already scheduled DR when editing 
image

File: src/backend/src/routes/design-reviews.routes.ts (hunk 1 above)
Comment (Peyton-McKee) [reply src/backend/src/routes/design-reviews.routes.ts:41]: i was submitting a url and it wasnt passing

**Comment by Peyton-McKee** [comment]
The I that appears for existing meetings I assume is because the design review Team Type Icon looks like that

Screenshot 2024-06-11 at 5 53 18 PM

**Review by walker-sean — APPROVED** [review]
LGTM

**Pull request merged**